dev/NavigationView/NavigationView.cpp

```diff
@@ -3331,6 +3402,54 @@ void NavigationView::UpdateListViewItemSource()
     }
 }
 
+void NavigationView::UpdateFooterListViewItemSource()
+{
+    auto footerItems = FooterMenuItems();
+
+    // If the old settings item is selected, move the selection to the new one.
+    auto selectedItem = SelectedItem();
+    bool shouldSelectSetting = selectedItem && IsSettingsItem(selectedItem);
+
+    if (shouldSelectSetting)
+    {
+        SetSelectedItemAndExpectItemInvokeWhenSelectionChangedIfNotInvokedFromAPI(nullptr);
+    }
+
+    if (m_settingsItem)
+    {
+        uint32_t index = 0;
+        if (footerItems.IndexOf(m_settingsItem.get(), index))
```

Comment:
>.get() [](start = 46, length = 6)

calling a tracker_ref's .get() method isn't free so we should try to only do it once.

I suggest that the if block be `if (auto&& settingsItem = m_settingsItem.get())`